apps/user-service/src/users/user-creation.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Injectable, NotFoundException, LoggerService, UnprocessableEntityException } from "@nestjs/common";
+import { JwtService } from "@nestjs/jwt";
+import { ModelHasRole, Role, User, Verification } from "@terramatch-microservices/database/entities";
+import { EmailService } from "@terramatch-microservices/common/email/email.service";
+import { LocalizationService } from "@terramatch-microservices/common/localization/localization.service";
+import { TMLogService } from "@terramatch-microservices/common/util/tm-log.service";
+import { UserNewRequest } from "./dto/user-new-request.dto";
+import crypto from "node:crypto";
+import { TemplateService } from "@terramatch-microservices/common/email/template.service";
+import { omit } from "lodash";
+import bcrypt from "bcryptjs";
+
+@Injectable()
+export class UserCreationService {
+  protected readonly logger: LoggerService = new TMLogService(UserCreationService.name);
+
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly emailService: EmailService,
+    private readonly templateService: TemplateService,
+    private readonly localizationService: LocalizationService
+  ) {}
+
+  async createNewUser(request: UserNewRequest): Promise<User> {
+    const bodyKey = "user-verification.body";
+    const subjectKey = "user-verification.subject";
+    const titleKey = "user-verification.title";
+    const ctaKey = "user-verification.cta";
+
+    const localizationKeys = await this.localizationService.getLocalizationKeys([
+      bodyKey,
+      subjectKey,
+      titleKey,
+      ctaKey
+    ]);
+
+    if (!localizationKeys.length) {
+      throw new NotFoundException("Localizations not found");
+    }
+
+    const bodyLocalization = localizationKeys.find(x => x.key == bodyKey);
+    const subjectLocalization = localizationKeys.find(x => x.key == subjectKey);
+    const titleLocalization = localizationKeys.find(x => x.key == titleKey);
+    const ctaLocalization = localizationKeys.find(x => x.key == ctaKey);
+
+    if (bodyLocalization == null) {
+      throw new NotFoundException("Localization body not found");
+    }
+
+    if (subjectLocalization == null) {
+      throw new NotFoundException("Localization subject not found");
+    }
+
+    if (titleLocalization == null) {
+      throw new NotFoundException("Localization title not found");
+    }
+
+    if (ctaLocalization == null) {
+      throw new NotFoundException("Localization CTA not found");
+    }
+
+    const role = request.role;
+    const roleEntity = await Role.findOne({ where: { name: role } });
+
+    if (roleEntity == null) {
+      throw new NotFoundException("Role not found");
+    }
+
+    const userExist = await User.findOne({ where: { emailAddress: request.emailAddress } });
```

Comment:
This would be more efficient with `count()`: 
```
const userExists = (await User.count({ where: { emailAddress: request.emailAddress } })) !== 0;
```